hr_timesheet_retroactive_cost/wizards/hr_employee_timesheet_cost_wizard.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2022-2024 Moduon Team S.L. <info@moduon.team>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class HrEmployeeTimesheetCost(models.TransientModel):
+    _name = "hr.employee.timesheet.cost.wizard"
+    _description = "Employee timesheet cost wizard"
+
+    employee_id = fields.Many2one(comodel_name="hr.employee")
+    currency_id = fields.Many2one("res.currency")
+    timesheet_cost = fields.Monetary(currency_field="currency_id")
+    retroactive_cost_date = fields.Date(
+        help="Change de timesheet cost from this date to now.",
+        default=fields.Datetime.now,
+    )
+
+    def update_employee_retroactive_cost(self):
+        """Recalculates employee timesheet cost from the given date.
+
+        This method uses the base function _timesheet_postprocess from
+        account_analytic_line (hr_timesheet) to recalculate employee costs.
+        Finally logs cost changes in cost history model.
+        """
+        self.ensure_one()
+        self.employee_id.write({"timesheet_cost": self.timesheet_cost})
```

Comment:
**suggestion(performance):** batch this write with new cost creation:

```suggestion
        bad_costs = self.employee_id.timesheet_cost_history_ids.filtered_domain(
            [("starting_date", ">=", self.retroactive_cost_date)]
        )
        self.employee_id.sudo().write({
            "timesheet_cost": self.timesheet_cost,
            "timesheet_cost_history_ids": [
                fields.Command.set(self.employee_id.timesheet_cost_history_ids - bad_costs),
                fields.Command.create({
                    "employee_id": self.employee_id.id,
                    "currency_id": self.currency_id.id,
                    "timesheet_cost": self.timesheet_cost,
                    "retroactive_cost_date": self.retroactive_cost_date,
                }),
            ],
        })
```

This has some benefits:
- You can remove `_update_employee_history_cost` from `hr.employee.timesheet.cost`.
- It lets the ORM reduce the number of CRUD operations as much as possible.